ttnn/core.py

```diff
@@ -1379,4 +1379,22 @@ def mean(input_tensor: ttnn.Tensor, dim: Union[int, Tuple[int]], keepdim: bool =
     return output_tensor
 
 
+def _torch_average_pool_2d(input_tensor: ttnn.Tensor):
+    import torch
+
+    output_size = (1, 1)
+    input_tensor = ttnn.from_device(input_tensor)
+    input_tensor = ttnn.to_layout(input_tensor, ttnn.ROW_MAJOR_LAYOUT)
+    input_tensor = ttnn.to_torch(input_tensor)
+
+    return torch.nn.AdaptiveAvgPool2d(output_size)(input_tensor)
+
+
+@decorate_operation(torch_function=_torch_average_pool_2d)
+def average_pool_2d(input_tensor: ttnn.Tensor) -> ttnn.Tensor:
```

Comment:
Please rename this to `average_pool2d` to be the same as `torch`

Reply:
done!